ext/test/http/curl_http_test.cc

```diff
@@ -184,9 +185,10 @@ TEST_F(BasicCurlHttpTests, HttpResponse)
 TEST_F(BasicCurlHttpTests, SendGetRequest)
 {
   received_requests_.clear();
-  curl::SessionManager session_manager;
+  auto session_manager = http_client::HttpClientFactory::Create();
+  EXPECT_TRUE(session_manager != nullptr);
 
-  auto session = session_manager.CreateSession("127.0.0.1", HTTP_PORT);
+  auto session = session_manager->CreateSession("127.0.0.1", HTTP_PORT);
   auto request = session->CreateRequest();
```

Comment:
I was thinking more about the [Javascript model](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Using_XMLHttpRequest) of it, where your request (handled by whatever client) is not bound to destination URL until you actually tell it to open / fire for a given URL, e.g.
```javascript
var oReq = new XMLHttpRequest();
oReq.addEventListener("load", reqListener);
oReq.open("GET", "http://www.example.org/example.txt");
oReq.send();
```

That way the HTTP client may provide concrete implementation, that is not session-aware. Then you pass the `HttpRequest` object to it, that object contains the destination URL detail. Then HTTP client executes your request with given params.. In that model you do not need a session manager.. Unless you want the session manager to handle some additional details, such as authentication?

Reply:
`SessionManager` handles multiple `Session`. I understand we can get rid of `Session` object, and directly use `Request` object . But this PR changes still hold true as this return `SessionManager` object specific to client implementation.